src/listbox/Listbox.ts

```diff
@@ -0,0 +1,264 @@
+ * @property getOptionDisabled    Function that accepts option data and returns a boolean for disabled/not disabled
+ * @property getOptionId          Function that accepts option data and returns a string ID
+ * @property getOptionSelected    Function that accepts option data and returns a boolean for selected/unselected
+ * @property onActiveIndexChange  Called with the index of the new requested active descendant
+ * @property onOptionSelect       Called with the option data of the new requested selected item
+ */
+
+export interface ListboxProperties extends ThemeableProperties {
+	activeIndex?: number;
+	describedBy?: string;
+	visualFocus?: boolean;
+	id?: string;
+	multiselect?: boolean;
+	optionData?: any[];
+	tabIndex?: number;
+	getOptionDisabled?(option: any, index: number): boolean;
+	getOptionId?(option: any, index: number): string;
+	getOptionLabel?(option: any, index: number): DNode;
+	getOptionSelected?(option: any, index: number): boolean;
+	onActiveIndexChange?(index: number, key: string | number): void;
+	onOptionSelect?(option: any, index: number, key: string | number): void;
+	onKeyDown?(event: KeyboardEvent, key: string | number): void;
+}
+
+export const ListboxBase = ThemeableMixin(WidgetBase);
+
+@theme(css)
+@diffProperty('optionData', reference)
+export default class Listbox extends ListboxBase<ListboxProperties> {
+	private _scroll: number | undefined;
+	private _idBase = uuid();
+
+	private _getOptionId(index: number): string {
+		const { optionData = [], getOptionId } = this.properties;
+		return getOptionId ? getOptionId(optionData[index], index) : `${this._idBase}-${index}`;
+	}
+
+	private _onKeyDown(event: KeyboardEvent) {
+		const {
+			activeIndex = 0,
+			key = '',
+			optionData = [],
+			getOptionDisabled = (option: any, index: number) => false,
+			onActiveIndexChange,
+			onOptionSelect,
+			onKeyDown
+		} = this.properties;
+
+		onKeyDown && onKeyDown(event, key);
+
+		const activeItem = optionData[activeIndex];
+		let newIndex: number;
+
+		switch (event.which) {
+			case Keys.Enter:
+			case Keys.Space:
+				event.preventDefault();
+				if (!getOptionDisabled(activeItem, activeIndex)) {
+					onOptionSelect && onOptionSelect(activeItem, activeIndex, key);
+				}
+				break;
+			case Keys.Down:
+				event.preventDefault();
+				newIndex = (activeIndex + 1) % optionData.length;
+				onActiveIndexChange && onActiveIndexChange(newIndex, key);
+				break;
+			case Keys.Up:
+				event.preventDefault();
+				newIndex = (activeIndex - 1 + optionData.length) % optionData.length;
+				onActiveIndexChange && onActiveIndexChange(newIndex, key);
+				break;
+			case Keys.Home:
+				onActiveIndexChange && onActiveIndexChange(0, key);
+				break;
+			case Keys.End:
+				onActiveIndexChange && onActiveIndexChange(optionData.length - 1, key);
+				break;
+		}
+	}
+
+	protected animateScroll(element: HTMLElement, scrollValue: number) {
+		element.scrollTop = scrollValue;
```

Comment:
I think the correct way to do this would be to create a local `scrollTop` meta within this file that takes a `key` and a `scrollValue`. That would avoid the need to use `onElementCreated` / `onElementUpdated`